scripts/gen_csr_test.py

```diff
@@ -35,325 +34,362 @@
+        "Please install bitstring package: sudo apt-get install python3-bitstring")
+    sys.exit(RET_FAIL)
 
 """
 Defines the test's success/failure values, one of which will be written to
 the chosen signature address to indicate the test's result.
 """
 TEST_RESULT = 1
-TEST_PASS   = 0
-TEST_FAIL   = 1
+TEST_PASS = 0
+TEST_FAIL = 1
+
 
 def get_csr_map(csr_file, xlen):
-  """
-  Parses the YAML file containing CSR descriptions.
-
-  Args:
-    csr_file: The CSR YAML file.
-    xlen: The current RISC-V ISA bit length.
-
-  Returns:
-    A dictionary contining mappings for each CSR, of the form:
-    { csr_name : [csr_address, csr_val_bitarray, csr_write_mask_bitarray, csr_read_mask_bitarray] }
-  """
-  rv_string = "rv{}".format(str(xlen))
-  csrs = {}
-  with open(csr_file, "r") as c:
-    csr_description = yaml.safe_load(c)
-    for csr_dict in csr_description:
-      csr_name = csr_dict.get("csr")
-      csr_address = csr_dict.get("address")
-      assert(rv_string in csr_dict), "The {} CSR must be configured for rv{}".format(csr_name, str(rv))
-      csr_value = bitarray(uintbe=0, length=xlen)
-      csr_write_mask = []
-      csr_read_mask = bitarray(uintbe=0, length=xlen)
-      csr_field_list = csr_dict.get(rv_string)
-      for csr_field_detail_dict in csr_field_list:
-        field_type = csr_field_detail_dict.get("type")
-        field_val = csr_field_detail_dict.get("reset_val")
-        field_msb = csr_field_detail_dict.get("msb")
-        field_lsb = csr_field_detail_dict.get("lsb")
-        field_size = field_msb - field_lsb + 1
-        if field_type != "WPRI":
-          val_bitarray = bitarray(uint=field_val, length=field_size)
-          mask_bitarray = bitarray(uint=1, length=1) * field_size
-          start_pos = xlen - 1 - field_msb
-          end_pos = xlen - 1 - field_lsb
-          csr_read_mask.overwrite(mask_bitarray, xlen - 1 - field_msb)
-          csr_value.overwrite(val_bitarray, xlen - 1 - field_msb)
-          access = True if field_type == "R" else False
-          csr_write_mask.append([mask_bitarray, (start_pos, end_pos), access])
-      csrs.update({csr_name : [csr_address, csr_value, csr_write_mask, csr_read_mask]})
-  return csrs
+    """
+    Parses the YAML file containing CSR descriptions.
+
+    Args:
+      csr_file: The CSR YAML file.
+      xlen: The current RISC-V ISA bit length.
+
+    Returns:
+      A dictionary contining mappings for each CSR, of the form:
+      { csr_name : [csr_address, csr_val_bitarray, csr_write_mask_bitarray, csr_read_mask_bitarray] }
+    """
+    rv_string = "rv{}".format(str(xlen))
+    csrs = {}
+    with open(csr_file, "r") as c:
+        csr_description = yaml.safe_load(c)
+        for csr_dict in csr_description:
+            csr_name = csr_dict.get("csr")
+            csr_address = csr_dict.get("address")
+            assert (
+                        rv_string in csr_dict), "The {} CSR must be configured for rv{}".format(
+                csr_name, str(rv))
```

Comment:
is it possible to fix this formatting? I think it looks a bit strange like this

Reply:
Good catch, I fixed this.